### ISSUE TITLE
Investigate alternative to the global app's 'dance'

### ISSUE THREAD

**Issue by yanick** [PerlDancer/Dancer2#1051]
> We're basically adding hints that another loop around the
> different App dispatchers can recognize and change things
> appropriately, and each App dispatcher decides whether to add hints or
> not, depending on whether it recognizes that it runs in `dance` loop
> mode.

Must try to see if there's a less convoluted way of doing it.

**Comment by yanick** [comment]
>>  On that one I think I'm with
>> you for `dance` having a lot of flailing and duct-taping happening
>> behind the curtain, and it'd be better to call a namespace's app
>> directly. But I'm unclear on the exact form that would take.
> 
> This is relatively easy. When we moved the dispatchers to the App
> itself (so each App is in charge of maintaining the loop around its
> routes), all you need to do is call "to_app" on the class.
> 
>   package MyApp {
>     use Dancer2;
>     # ... routes ...
>   }
> 
>   # app.psgi
>   use MyApp;
>   MyApp->to_app;
> 
> If you want to mix more than one "Dancer Application", you can just
> use Plack::Builder
> 
>   # app.psgi
>   use MyApp;
>   use MyApp::API; # <- might have a serializer
>   use Plack::Builder;
> 
>   builder {
>     # order doesn't matter, it does a reverse sort on length
>     mount '/api' => MyApp::API->to_app; # as the API
>     mount '/' => MyApp->to_app; # as the main app
>   };

**Comment by xsawyerx** [comment]
I'm not sure what this ticket is meant to highlight. We have a better alternative to the `dance` keyword. The problem I described in emails is how we currently support the `dance` keyword to make it still work the way it should despite it being a bad practice.

What I want is to have to go through less hoops in order to support the `dance` keyword, or to remove it entirely. Does this relate to the suggested `dancefloor` keyword?

Perhaps we should describe this ticket better.

**Comment by yanick** [comment]
>  Does this relate to the suggested dancefloor keyword?

Yes. The main point of the ticket is a ntoe for myself to look and see what we're doing right now, and see if/how I can improve on it.
